### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#141 

Builds and tests pass locally. Need to check travis and appveyor.

### CODE REVIEW

**Comment by natemcmaster** [comment]
You'll need to update appveyor.yml to include `os: Visual Studio 2017 RC`

File: src/Microsoft.AspNetCore.WebSockets/Microsoft.AspNetCore.WebSockets.csproj [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageTags>aspnetcore</PackageTags>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Http.Extensions" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Options" Version="1.2.0-*" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.WebSockets/Microsoft.AspNetCore.WebSockets.csproj]: This whole ItemGroup should be removed. It's not necessary to compile for net451

File: test/Microsoft.AspNetCore.WebSockets.ConformanceTest/Microsoft.AspNetCore.WebSockets.ConformanceTest.csproj [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Server.IntegrationTesting" Version="0.3.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.Extensions.PlatformAbstractions" Version="1.2.0-*" />
+    <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.4.0-*" />
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.WebSockets.ConformanceTest/Microsoft.AspNetCore.WebSockets.ConformanceTest.csproj]: Nit sort

File: test/Microsoft.AspNetCore.WebSockets.Test/Microsoft.AspNetCore.WebSockets.Test.csproj [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="1.2.0-*" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-*" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0-*" />
+    <PackageReference Include="xunit" Version="2.2.0-*" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.WebSockets.Test/Microsoft.AspNetCore.WebSockets.Test.csproj]: Diddo. Not required

File: src/Microsoft.AspNetCore.WebSockets/Microsoft.AspNetCore.WebSockets.csproj (hunk 1 above)
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.WebSockets/Microsoft.AspNetCore.WebSockets.csproj]: Yea I'm not sure where this came from but I think the dotnet migrate added it so I kept it.

File: samples/EchoApp/EchoApp.csproj [hunk 4]
```diff
@@ -4,7 +4,6 @@
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <PreserveCompilationContext>true</PreserveCompilationContext>
     <OutputType>Exe</OutputType>
-    <PackageTargetFallback>$(PackageTargetFallback);dotnet5.6;portable-net45+win8</PackageTargetFallback>
```
Comment (natemcmaster) [inline samples/EchoApp/EchoApp.csproj]: 🎉

**Comment by natemcmaster** [comment]
:shipit: when CI passes. We may need to fix aspnet/Korebuild#162 to make xplat build work.

**Pull request merged**